modules/base/manifests/init.pp

```diff
@@ -2,24 +2,38 @@
 #
 #
 
-class base {
+class base (
+	$period 	 	 = 'weekly',
+	$repeat 	 	 = '1'
+){
+	# schedule parameters for update
+	schedule { 'update':
+ 		period 		 => $period,
+	  	repeat   	 => $repeat,
+	}
 
-if $::osfamily == 'RedHat' {
-  exec { 'yum-update':
-      command      => '/usr/bin/yum -y update'
-      }
-}
-
-elsif $::osfamily == 'Debian' {
-
-    exec { 'apt-update':
-      command      => '/usr/bin/apt-get -y update'
-      }
+	# OS parameters for Update and ntp-service
+	if $::osfamily == 'RedHat' {
+		$ntp_service = 'ntpd'
+		exec { '/usr/bin/yum -y update':
+ 	 	schedule 	 => 'update',
+ 		}
+	}
+	elsif $::osfamily == 'Debian' {
+		$ntp_service = 'ntp'
+		exec { '/usr/bin/apt-get -y update':
+ 	 	schedule 	 => 'update',
+ 		}
+	}
 
-}
-
-$packages = ['tree', 'mc', 'net-tools']
-package { $packages:
-    ensure       => installed,
-}
+	# Start packages
+	$packages = ['tree', 'mc', 'net-tools', 'ntp']
+	package { $packages:
+		ensure       => installed,
+	}	
+	# Configure ntp service
+	class { 'base::ntp':
+		ntp_service	 => $ntp_service,
+		require      => Package[$packages],	
```

Comment:
I dont understand why ntp required packages  $packages = ['tree', 'mc', 'net-tools',

Reply:
The same question.